accepted/00000-modular-repos-with-clm.md

```diff
@@ -0,0 +1,181 @@
+- Feature Name: Modular Repositories with CLM
+- Start Date: 2019-10-23
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Implement additional functionality on top of content lifecycle management to be able to build regular repositories out of modular repositories.
+
+Design additional CLM "module" filters to let the user pick desired modules and streams.
+
+# Motivation
+[motivation]: #motivation
+
+RHEL/CentOS 8 inherits a new concept of [modularity and modular repositories](https://docs.fedoraproject.org/en-US/modularity/).
+Modularity allows different major versions (streams) to coexist in a single repository.
+Users enable/disable streams so that there is only a single stream of a module is visible to the package manager at once.
+
+Modular repositories are defined in a new type of metadata called `modulemd` in a file named `modules.yaml`. This metadata has to be interpreted properly to manage these repositories.
+Unless this metadata is handled, managing modular repositories causes problems with package operations, specifically, suggesting incorrect package updates in the UI.
+
+Currently, Uyuni has only the most basic support for modularity by displaying the following message in the package management UI:
+```
+At least one of the channels this system is subscribed to contains modules. If you have activated modules on this system, please refrain from using Spacewalk for package operations. Instead, perform all package actions from the client using dnf.
+```
+
+The goal is to handle modular repositories in a way that the clients can be still managed through the Uyuni UI.
+
+# Detailed design
+[design]: #detailed-design
+
+## Overview
+
+Instead of implementing the modularity logic inside the core of Uyuni, this document describes a design to simulate the modularity behavior by building regular repositories out of modular repositories by filtering out packages from unselected streams.
```

Comment:
Not much. in the general sense, the concept of modularity is directly related to a repository so that's why I decided to go with it.

Reply:
Hmm, that's true as well. I asked because I was confused - at first read it might seem the RFC would result in a feature that has repos as an output (and then users have to create channels out of them), while actually knowing the CLM feature I am sure channels is what comes out at the end.

Feel free to either ignore or clarify if you find a way to. In the end this is nothing substantial really, so I am marking this as resolved.

Thanks!